src/XRP/model/xrp-deposit-preauth.ts

```diff
@@ -27,6 +28,13 @@ export default class XrpDepositPreauth {
       .getUnauthorize()
       ?.getValue()
       ?.getAddress()
+    // You must provide either Authorize or Unauthorize, but not both.
+    if (authorize && unauthorize) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'DepositPreauth protobuf provides both `Authorize` and `Unauthorize` fields.',
+      )
+    }
```

Comment:
Not covered in tests because it appears that it's not actually reachable, since `DepositPreauth` doesn't let both `authorize` and `unauthorize` be set together.

Reply:
oh interesting - if you setup a protobuf with both fields set, it fails?

Reply:
No, the second one you set just nulls the first one.